File: IBTextRange/ViewController.m
//
//  ViewController.m
//  IBTextRange
//
//  Created by ET|冰琳 on 16/11/17.
//  Copyright © 2016年 IB. All rights reserved.
//

#import "ViewController.h"
#import "UITextView+textRange.h"
#import "UITextField+textRange.h"

@interface ViewController ()

@property (nonatomic, weak) IBOutlet UILabel *textFieldRemainTextLabel;
@property (nonatomic, weak) IBOutlet UITextField *textField;
@property (nonatomic, weak) IBOutlet UILabel *textViewRemainTextLabel;
@property (nonatomic, weak) IBOutlet UITextView *textView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
   [self.textView ib_observerTextLengthChanged:^(NSInteger currentLength) {
       self.textViewRemainTextLabel.text = [NSString stringWithFormat:@"UITextView字数限制 (%@)", @(10 - currentLength)];
   }];
    
    [self.textField ib_observerTextLengthChanged:^(NSInteger currentLength) {
        self.textViewRemainTextLabel.text = [NSString stringWithFormat:@"UITextView字数限制 (%@)", @(10 - currentLength)];
    }];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: IBTextRange/ViewController.swift
//
//  ViewController.swift
//  UITextInputConstraits
//
//  Created by ET|冰琳 on 16/2/25.
//  Copyright © 2016年 Ice Butterfly. All rights reserved.
//

import UIKit

class ViewController: UIViewController{
    
    @IBOutlet weak var field: UITextField!
    @IBOutlet weak var textView: UITextView!

    @IBOutlet weak var textViewRemainLabel: UILabel!
    @IBOutlet weak var textfieldRemainText: UILabel!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.field.text = "123"
        self.textView.text = "1"
        
        self.textView.observerTextLengthChanged {[unowned self] (le) in
            print("lenght change")
            self.textViewRemainLabel.text = "(" + "\(self.textView.getRemainTextLength())" + ")"
            
        }
        self.field.observerTextLengthChanged {[unowned self] (len) in
            self.textfieldRemainText.text = "(" + "\(self.field.getRemainTextLength())" + ")"
        }
        
        self.textView.placeholder = "请输入"
        self.textView.maxLength = 30;
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

